draft-ietf-jsonpath-base.md

```diff
@@ -422,25 +422,42 @@ type errors when pondering well-formedness and validity, while
 resource depletion and related errors are comparable to 500 type
 errors.)
 
-## Processing Model
+## Syntax
+
+Syntactically, a JSONPath query consists of a root selector (`$`), which
+stands for a nodelist that contains the root node of the argument,
+followed by a possibly empty sequence of *selectors*.
+
+~~~~ abnf
+json-path = root-selector *(S (dot-selector        /
+                               dot-wild-selector   /
+                               index-selector      /
+                               index-wild-selector /
+                               slice-selector      /
+                               descendant-selector /
+                               filter-selector     /
+                               list-selector))
+~~~~
+
+The syntax and semantics of each selector is defined below.
 
-In this specification, the semantics of a JSONPath query are defined
-in terms of a *processing model*.  That model is not prescriptive of
-the internal workings of an implementation:  Implementations may wish
-(or need) to design a different process that yields results that are
-consistent with this model.
```

Comment:
I think the "consistent with this model" sentence is fairly important to keep.  We're still defining a semantic "how it works."  It's important to say that implementations can do what they want so long as the output is what is expected.

Reply:
@gregsdennis Do you think "consistent with this model" needs adding, given the following opening sentence of this section?

> In this specification, the semantics of a JSONPath query define the
required results and do not prescribe the internal workings of an
implementation.

Reply:
Ah. I think that's fine.